tests/unit/multi-currency/compatibility/test-class-woocommerce-product-add-ons.php

```diff
@@ -449,7 +461,10 @@ public function test_get_item_data_returns_custom_price_data_correctly() {
 			'display' => ' (<span class="woocommerce-Price-amount amount"><bdi><span class="woocommerce-Price-currencySymbol">&#36;</span>42.00</bdi></span>)',
 		];
 
-		$this->assertSame( $expected, $this->woocommerce_product_add_ons->get_item_data( [], $addon, $cart_item ) );
+		$actual = $this->woocommerce_product_add_ons->get_item_data( [], $addon, $cart_item );
+		$this->assertSame( $expected['name'], $actual['name'] );
+		$this->assertStringContainsString( '$42.00', html_entity_decode( strip_tags( $actual['value'] ), ENT_QUOTES | ENT_HTML5, 'UTF-8' ) );
+		$this->assertStringContainsString( '$42.00', html_entity_decode( strip_tags( $actual['display'] ), ENT_QUOTES | ENT_HTML5, 'UTF-8' ) );
```

Comment:
This change is a bit aggressive. I think we should keep validating the whole `value` and `display`, and just focus on strip_tags in this PR. So we just need to adjust `value` and `display` values to fix this issue, together with my idea of using a util `array_strip_tags`.

Reply:
Much better, thanks. Addressed here [44be7fe](https://github.com/Automattic/woocommerce-payments/pull/11034/commits/44be7fe3ce4284bfc86ab6cced30960ab84cb828) as well